0.Preprocess_Data/scripts/2.preprocessing_nELISA_data.py

```diff
@@ -0,0 +1,326 @@
+].str.replace("µ", "u")
+# replace all "nan" with "0"
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("nan", "0")
+
+
+# In[ ]:
+
+
+# _0.03 to _0.025
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_0.03", "_0.025", regex=False)
+# _0.010_ to _0.100_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_0.01_", "_0.010_", regex=False)
+# _0.10_ to _0.100_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_0.10_", "_0.100_", regex=False)
+# _0.10_ to _0.100_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_0.1_", "_0.100_", regex=False)
+# _1_ to _1.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_1_", "_1.000_", regex=False)
+# _1.0_ to _1.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_1.0_", "_1.000_", regex=False)
+# _10_ to _10.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_10_", "_10.000_", regex=False)
+# _100_ to _100.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_100_", "_100.000_", regex=False)
+# _100.0_ to _100.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_100.0_", "_100.000_", regex=False)
+# _5_ to _5.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_5_", "_5.000_", regex=False)
+# _20_ to _20.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_20_", "_20.000_", regex=False)
+# _30_ to _30.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_30_", "_30.000_", regex=False)
+# _2.5 to _2.500_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_2.5_", "_2.500_", regex=False)
+# _3.0 to _3.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_3.0_", "_3.000_", regex=False)
+# _3 to _3.000_
+analysis_df["oneb_Treatment_Dose_Inhibitor_Dose"] = analysis_df[
+    "oneb_Treatment_Dose_Inhibitor_Dose"
+].str.replace("_3_", "_3.000_", regex=False)
+
+
+# In[ ]:
+
+
+# save nELISA_plate_430420_no_inhibitors dataframe to csv file
+analysis_df.to_csv(preprocessing_path, index=False)
```

Comment:
Are there reasons for using CSV instead of parquet here? Mostly I ask out of curiosity. There may be a benefit to sticking with one format consistently for any outputs, depending on your constraints.

Reply:
changed to parquet